RecoHGCal/TICL/interface/PatternRecognitionAlgoBase.h

```diff
@@ -34,20 +34,18 @@ namespace ticl {
       const edm::ValueMap<float>& layerClustersTime;
       const TICLLayerTiles& tiles;
       const std::vector<TICLSeedingRegion>& regions;
-      std::vector<Trackster>& result;
 
       Inputs(const edm::Event& eV,
              const edm::EventSetup& eS,
              const std::vector<reco::CaloCluster>& lC,
              const std::vector<float>& mS,
              const edm::ValueMap<float>& lT,
              const TICLLayerTiles& tL,
-             const std::vector<TICLSeedingRegion>& rG,
-             std::vector<Trackster>& rS)
-          : ev(eV), es(eS), layerClusters(lC), mask(mS), layerClustersTime(lT), tiles(tL), regions(rG), result(rS) {}
+             const std::vector<TICLSeedingRegion>& rG)
+          : ev(eV), es(eS), layerClusters(lC), mask(mS), layerClustersTime(lT), tiles(tL), regions(rG) {}
     };
 
-    virtual void makeTracksters(const Inputs& input) = 0;
+    virtual const void makeTracksters(const Inputs& input, std::vector<Trackster>& result) = 0;
```

Comment:
I meant the method qualifier, not the return value qualifier.
```virtual void makeTracksters(const Inputs& input) const = 0;```

Reply:
with this change (const as method qualifier, not return value qualifier), it arise the compilation error 
RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc:26:34:  error: passing 'const hgcal::RecHitTools' as 'this' argument discards qualifiers [-fpermissive]
   rhtools_.getEventSetup(input.es);

Is there an easy fix, or should I remove the const?

Reply:
I see. So, it looks like at least for some of the derived classes the methods are not reentrant
and we can not make it const (or thread safe).
It seems fine now to just revert this attempt to add a const.

Reply:
ok!